sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3869,6 +3868,24 @@ class AstBuilder extends DataTypeAstBuilder
    * DataType parsing
    * ******************************************************************************************** */
 
+  override def visitSingleRoutineParamList(
+      ctx: SingleRoutineParamListContext): StructType = withOrigin(ctx) {
+    val schema = createSchema(ctx.colDefinitionList())
+    val fields = for (field <- schema) yield {
+      field.getCurrentDefaultValue() match {
```

Comment:
shall we add assert here that only comment and default value can be present? This is guaranteed by the parser.